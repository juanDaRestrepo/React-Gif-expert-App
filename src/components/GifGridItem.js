import React from 'react'

export const GifGridItem = ({id,url,title}) => {
    
    return (
        <div className='card animate__animated animate__backInLeft '>
            <img src={url} alt={title}/>
            <p>{title}</p>
        </div>
    )
}
