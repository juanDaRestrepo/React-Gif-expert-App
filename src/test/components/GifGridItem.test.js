import {shallow} from 'enzyme';
import React from 'react';
import {GifGridItem} from '../../components/GifGridItem';

describe('Pruebas en <GifGridItem />', () => {
        const url = "http://algo";
        const title = "algun titulo";
        const wrapper = shallow(<GifGridItem title={title} url={url}/>);
    test('debe de mostrar el componente correctamente', () => {
        
        
        expect(wrapper).toMatchSnapshot();
    })

    test('Debe contener un parrafo con el title', () => {
        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(title);
    })

    test('Debe de tener la imagen igual al url y alt de los props', () => {
        const img = wrapper.find('img');
        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);
    })
    
    test('div de la card debe tener las clases animate__animated animate__backInLeft', () => {
        const div = wrapper.find('div')
        console.log(div.prop('className'));
        const classN = div.prop('className')
        expect(classN.includes('animate__backInLeft')).toBe(true)
    })
    
})

