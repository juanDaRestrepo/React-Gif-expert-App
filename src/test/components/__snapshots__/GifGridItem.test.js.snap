// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem /> debe de mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__backInLeft "
>
  <img
    alt="algun titulo"
    src="http://algo"
  />
  <p>
    algun titulo
  </p>
</div>
`;
